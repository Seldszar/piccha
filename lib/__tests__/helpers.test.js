const helpers = require("../helpers");

describe("toPath()", () => {
  test("should handle arrays", () => {
    expect(helpers.toPath(["lorem", "ipsum"])).toMatchSnapshot();
  });

  test("should handle dot strings", () => {
    expect(helpers.toPath("lorem.ipsum")).toMatchSnapshot();
  });

  test("should handle slash strings", () => {
    expect(helpers.toPath("/lorem/ipsum")).toMatchSnapshot();
  });

  test("should convert other types into arrays", () => {
    expect(helpers.toPath({})).toMatchSnapshot();
  });
});

describe("mergePaths()", () => {
  test("should merge paths", () => {
    expect(helpers.mergePaths("/lorem", "ipsum")).toMatchSnapshot();
  });
});

describe("normalizePath()", () => {
  test("should normalize input", () => {
    expect(helpers.normalizePath("lorem.ipsum")).toMatchSnapshot();
  });
});

describe("ensureTrailingSlash()", () => {
  test("should add a trailing slash", () => {
    expect(helpers.ensureTrailingSlash("/lorem/ipsum")).toMatchSnapshot();
  });

  test("should not add a trailing slash", () => {
    expect(helpers.ensureTrailingSlash("/lorem/ipsum/")).toMatchSnapshot();
  });
});

describe("mapNodeValues()", () => {
  test("should fallback to an empty array", () => {
    expect(helpers.mapNodeValues(null)).toMatchSnapshot();
  });

  test("should handle objects", () => {
    expect(helpers.mapNodeValues({ "/lorem": "ipsum" })).toMatchSnapshot();
  });

  test("should handle array pairs", () => {
    expect(helpers.mapNodeValues([["/lorem", "ipsum"]])).toMatchSnapshot();
  });
});
