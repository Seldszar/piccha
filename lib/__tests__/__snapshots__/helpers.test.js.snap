// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensureTrailingSlash() should add a trailing slash 1`] = `"/lorem/ipsum/"`;

exports[`ensureTrailingSlash() should not add a trailing slash 1`] = `"/lorem/ipsum/"`;

exports[`mapNodeValues() should fallback to an empty array 1`] = `Array []`;

exports[`mapNodeValues() should handle array pairs 1`] = `
Array [
  Array [
    "/lorem",
    "ipsum",
  ],
]
`;

exports[`mapNodeValues() should handle objects 1`] = `
Array [
  Array [
    "/lorem",
    "ipsum",
  ],
]
`;

exports[`mergePaths() should merge paths 1`] = `
Array [
  "lorem",
  "ipsum",
]
`;

exports[`normalizePath() should normalize input 1`] = `"/lorem/ipsum"`;

exports[`toPath() should convert other types into arrays 1`] = `
Array [
  Object {},
]
`;

exports[`toPath() should handle arrays 1`] = `
Array [
  "lorem",
  "ipsum",
]
`;

exports[`toPath() should handle dot strings 1`] = `
Array [
  "lorem",
  "ipsum",
]
`;

exports[`toPath() should handle slash strings 1`] = `
Array [
  "lorem",
  "ipsum",
]
`;
